docs-src/concepts.md

```diff
@@ -230,7 +230,24 @@ be differences in behavior from the spec.
 
 ### Extension points
 
-TODO
+A number of extension points are provided in the API to provide flexibility in
```

Comment:
This PR has a lot of overlap with the additions to the FAQ in #386 by @jpeach, specifically the "How can I expose custom capabilities through Service API" question. I think that question/answer may be a bit too large for FAQ, maybe we can combine the content here?

Reply:
@jpeach what do you think makes the most sense here? Should we merge this one first and then add your content from FAQs in a follow up?